
# kp1/deseq2 analysis

## 1. prepare rsem output for deseq2

https://bioconductor.org/packages/3.7/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#transcript-abundance-files-and-tximport-input

http://bioconductor.org/packages/release/bioc/vignettes/tximport/inst/doc/tximport.html#rsem

check which R is compatible with tximport,
use if none compatible /users/lfresard/R_3.3.2/bin/R
```{r}
which R
module avail
```

```{bash}
module load r/3.4.2
R
```
```{r}
source("https://bioconductor.org/biocLite.R")
biocLite("tximport")
biocLite("readr")
```

load sample file
```{r}
dir <- "~/projects/kp1/"
samples <- read.table(file.path(dir, "sample", "kp1_sample_table.txt"), header=TRUE)
```

```{r}
library("tximport")
library("readr")
files <- file.path(dir, "rsem", "rsem_output", paste0(samples$rna_lib, ".Aligned.toTranscriptome.genes.results"))
names(files) <- paste0("sample", 1:21)
txi.rsem <- tximport(files, type = "rsem", txIn = FALSE, txOut = FALSE)
head(txi.rsem$counts)
```

## 2. deseq2 and p-values

## 3. ma plot, html report

https://bioconductor.org/packages/3.7/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#rich-visualization-and-reporting-of-results
http://bioconductor.org/packages/release/bioc/vignettes/pcaExplorer/inst/doc/pcaExplorer.html
http://bioconductor.org/packages/release/bioc/html/ReportingTools.html

## 4. pca among samples

https://bioconductor.org/packages/3.7/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#principal-component-plot-of-the-samples
https://bioconductor.org/packages/3.7/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#data-quality-assessment-by-sample-clustering-and-visualization

## 5. vst (variance stablizing transformation)

https://bioconductor.org/packages/3.7/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#variance-stabilizing-transformation





